### PR TITLE
update generate-octicons script

### LINKED ISSUE #2895: generate-octicons script should also prettify generated code
The example code in `script/is-it-pretty.js` can help to get this working.

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #2895

A new version of the `octicons` package was published a couple of weeks ago. They've finally done away with the unified SVG icon we used, so to keep up we need to change up our script a bit.

I decided to just enumerate the octicons returned from `require('octicons')` and extract the relevant bits there.

For reference, this is what you get through the JS API:

```
var octicons = require("octicons")
octicons.alert
// { keywords: [ 'warning', 'triangle', 'exclamation', 'point' ],
//   path: '<path d="M8.865 1.52c-.18-.31-.51-.5-.87-.5s-.69.19-.87.5L.275 13.5c-.18.31-.18.69 0 1 .19.31.52.5.87.5h13.7c.36 0 .69-.19.86-.5.17-.31.18-.69.01-1L8.865 1.52zM8.995 13h-2v-2h2v2zm0-3h-2V6h2v4z"/>',
//   height: '16',
//   width: '16',
//   symbol: 'alert',
//   options:
//    { version: '1.1',
//      width: '16',
//      height: '16',
//      viewBox: '0 0 16 16',
//      class: 'octicon octicon-alert',
//      'aria-hidden': 'true' },
//   toSVG: [Function] }
```

The big change here is that the order of keys is slightly different compared to the unified file, but aside from that everything else ported over fine. 

I've eyeballed the various icons I could find in the app after this change, they all seemed fine too.

The script now also prettifies the file after it's written, so you don't accidentally trip up `eslint`/`tslint`.

### CODE REVIEW

File: script/generate-octicons.ts [hunk 1]
```diff
@@ -1,38 +1,80 @@
...
-  '..',
-  'node_modules',
-  'octicons',
-  'build',
-  'sprite.octicons.svg'
-)
+interface IXML2JSNode {
+  path: {
+    $: {
+      d: string
```
Comment (joshaber) [inline script/generate-octicons.ts]: Heckuva path.

Comment (shiftkey) [reply script/generate-octicons.ts]: T Y P E S

**Comment by shiftkey** [comment]
> Should we be calling the `toSVG` function instead of creating the SVG tag ourselves? 

There's a few little things [we add inside our `Octicon` component](https://github.com/desktop/desktop/blob/master/app/src/ui/octicons/octicon.tsx#L65-L80) which overlap here, but setting a title doesn't seem to be supported.

This also means we'll take a runtime dependency on `octicons`, rather than our optimized fancy enums and component which Just Work™.

**Comment by joshaber** [comment]
>This also means we'll take a runtime dependency on octicons, rather than our optimized fancy enums and component which Just Work™.

Ah, that's fair. Merge conflict and then this'll be good 👍

**Comment by shiftkey** [comment]
:apple:

**Pull request merged**